exporter/opensearchexporter/model.go

```diff
@@ -0,0 +1,100 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package opensearchexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/opensearchexporter"
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/opensearchexporter/internal/objmodel"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/traceutil"
+)
+
+type mappingModel interface {
+	encodeLog(pcommon.Resource, plog.LogRecord) ([]byte, error)
+	encodeSpan(pcommon.Resource, ptrace.Span) ([]byte, error)
+}
+
+// encodeModel tries to keep the event as close to the original open telemetry semantics as is.
+// No fields will be mapped by default.
+//
+// Field deduplication and dedotting of attributes is supported by the encodeModel.
```

Comment:
What is dedotting?

Reply:
Have a look at exporter/opensearchexporter/internal/objmodel/objmodel.go. This explains a lot about it. In short, this is something, that is required for OpenSearch due to its interpretation of dots in field names (keys).